Add Time boundaries to Between date range

protich opened the pull request:
Add time boundaries to between data range dates and convert results to database timezone.

protich committed:
Add Time boundaries to Between date range

Add time boundaries to between data range dates and convert results to
database timezone.

protich commented:
Addresses issue #5155

protich commented on include/class.forms.php:
```diff
@@ -2415,15 +2415,17 @@ function getSearchQ($method, $value, $name=false) {
+            $left = $left->setTime(00, 00, 00);
+            $right = $right->setTime(23, 59, 59);
+            $left->setTimezone($tz)->setTimezone($dbtz);
+            $right->setTimezone($tz)->setTimezone($dbtz);
```
I'm not so sure we should convert time to user's timezone. 

I added it because - between widget doesn't provide for opportunity specify timezone but I'll be interested in y'alls feedback.

JediKev replied:
I think it's a good idea as the Date Picker is shown in the user's timezone. So if they chose between `11/11/2019` and `11/12/2019` in the date picker they would expect the system to show results in that exact timeframe in their timezone. I tested this pull and it works as expected. It pulls the correct data.

protich replied:
The issue is mute when it comes to the agent doing Advanced Search - the date picker already registers the user's timezone as part of the entry. So we don't actually need to convert to user's timezone in this case.

My concern is when `Between` criteria is used in a global queue. The saved values will be on the queue creator's timezone - forcing conversion to the agent timezone might not be the intended results -  I added it anyhow because `Period` floats the timezone, but it's not specific tho.

protich committed:
Remove user's timezone conversion

The date picker already has the user's timezone factored in to the
selection. Removing user's timezone allows for admin set queues to maintain
the original timezone selection.
